File: frontend/src/types/wallet.ts
export interface Wallet {
  id: string;
  wallet_address: string;
  chain_id: string | null;
  first_seen: string | null;
  last_seen: string | null;
  total_in: number;
  total_out: number;
  created_at: string;
}

export interface WalletScore {
  id: string;
  wallet_address: string;
  risk_score: number | null;
  fan_in_score: number | null;
  fan_out_score: number | null;
  velocity_score: number | null;
  exposure_score: number | null;
  scored_at: string;
}

File: backend/app/schemas/wallet.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class WalletCreate(BaseModel):
    wallet_address: str
    chain_id: Optional[str] = None
    first_seen: Optional[datetime] = None
    last_seen: Optional[datetime] = None
    total_in: float = 0
    total_out: float = 0


class WalletResponse(WalletCreate):
    id: str
    created_at: datetime

    model_config = {"from_attributes": True}
